core/trino-main/src/main/java/io/trino/operator/output/UnnestingPositionsAppender.java

```diff
@@ -96,7 +117,15 @@ public long getSizeInBytes()
 
     private void appendDictionary(IntArrayList positions, DictionaryBlock source)
     {
-        delegate.append(mapPositions(positions, source), source.getDictionary());
+        Block dictionary = source.getDictionary();
+        IntArrayList mappedPositions = mapPositions(positions, source);
```

Comment:
It's not straightforward to tell what is `mappedPositions`. Maybe rename it to `dictionaryPositions`